test/Cache.spec.ts

```diff
@@ -29,6 +29,19 @@ describe('Cache', () => {
       await fs.outputFile(cachePath, 'dummy data');
       expect(await cache.getPathForFileInCache(dummyUrl, 'test.txt')).toEqual(cachePath);
     });
+
+    it('should strip the hash and query params off the url', async () => {
+      const assetName = 'electron-v7.2.4-darwin-x64.zip-v7.2.4-darwin-x64.zip';
+      const nonStrippedUrl =
+        'https://github-production-release-asset-2e65be.s3.amazonaws.com/261648044/1a34b080-8f14-11ea-8d7e-01584eb03de2?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20200507%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20200507T022624Z&X-Amz-Expires=300&X-Amz-Signature=8b3098ae54a6acc52d30f93c0084c8b09ebc49214c108ed9ef067beb66ce6d54&X-Amz-SignedHeaders=host&actor_id=2036040&repo_id=261648044&response-content-disposition=attachment%3B%20filename%3Delectron-v7.2.4-darwin-x64.zip&response-content-type=application%2Foctet-stream';
+      const strippedUrl =
+        'httpsgithub-production-release-asset-2e65be.s3.amazonaws.com2616480441a34b080-8f14-11ea-8d7e-01584eb03de2';
+      const cachePath = path.resolve(cacheDir, strippedUrl, assetName);
+
+      await fs.outputFile(cachePath, 'dummy data');
+
+      expect(await cache.getPathForFileInCache(nonStrippedUrl, assetName)).toEqual(cachePath);
```

Comment:
Can we use `toMatchInlineSnapshot` here to avoid the hardcoded expected URL.  It'll give us better diffs and an easier update path in the future.